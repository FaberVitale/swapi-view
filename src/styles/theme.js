//@flow
import {
  yellow200 as primary1,
  yellow500 as primary2,
  grey400 as accent3
} from "material-ui/styles/colors";
import darkBaseTheme from "material-ui/styles/baseThemes/darkBaseTheme";
import getMuiTheme from "material-ui/styles/getMuiTheme";

const boxShadow =
  "0 1px 6px rgba(0, 0, 0, 0.12), 0 1px 4px rgba(0, 0, 0, 0.12)";

const theme = getMuiTheme(darkBaseTheme, {
  palette: {
    primary1Color: primary1,
    primary2Color: primary2,
    accent3Color: accent3,
    accent1Color: primary1,
    alternateTextColor: darkBaseTheme.palette.canvasColor
  },
  topBar: {
    backgroundColor: darkBaseTheme.palette.canvasColor,
    padding: 0,
    margin: 0,
    boxShadow: boxShadow
  },

  paper: {
    boxShadow
  },

  homePaper: {
    backgroundColor: "#000000"
  },

  extLink: {
    color: primary1,
    outlineColor: primary2
  },

  h1: {
    color: primary2,
    fontSize: 24
  },
  h2: {
    color: primary2
  },

  noResults: {
    color: "rgba(255, 255, 255, .5)"
  },

  autocomplete: {
    icon: {
      fill: primary1
    }
  },

  textField: {
    floatingLabelColor: "#fefefe",
    focusColor: primary2,
    borderColor: "#fefefe",
    hintColor: "rgba(255, 255, 255, .5)"
  },

  dropDownMenu: {
    accentColor: "#fefefe",
    focusColor: primary2
  },

  h3: darkBaseTheme.h4,

  p: {
    color: primary2
  },
  failedToLoad: {
    frown: {
      color: primary2,
      opacity: 0.1
    }
  },
  menuItem: {
    selectedTextColor: primary2
  }
});

export default theme;
