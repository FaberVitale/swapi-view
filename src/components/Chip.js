//@flow
import theme from "../styles/theme";
import React from "react";
type Props = {
  text: string
};

/* non interactive replacement of Mui chip */
class Chip extends React.PureComponent<Props> {
  render() {
    return (
      <div className="chip-container" style={theme.chip}>
        <span className="chip-span">
          <strong>{this.props.text}</strong>
        </span>
      </div>
    );
  }
}

export default Chip;
