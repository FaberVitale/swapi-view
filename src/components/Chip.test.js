import React from "react";
import Renderer from "react-test-renderer";
import theme from "../styles/theme";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import Chip from "./Chip";

describe("components/Chip", () => {
  const Comp = (
    <MuiThemeProvider muiTheme={theme}>
      <Chip text="test" />
    </MuiThemeProvider>
  );

  const tree = Renderer.create(Comp).toJSON();

  it("matches the snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
});
