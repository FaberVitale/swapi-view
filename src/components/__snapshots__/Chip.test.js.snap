// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Chip matches the snapshot 1`] = `
<div
  className="chip-container"
  style={
    Object {
      "backgroundColor": "rgb(72, 72, 72)",
      "deleteIconColor": "rgba(255, 255, 255, 0.26)",
      "fontSize": 14,
      "fontWeight": 400,
      "shadow": "0 1px 6px rgba(0, 0, 0, 0.12),
            0 1px 4px rgba(0, 0, 0, 0.12)",
      "textColor": "rgba(255, 255, 255, 0.87)",
    }
  }
>
  <span
    className="chip-span"
  >
    <strong>
      test
    </strong>
  </span>
</div>
`;
